pagination/helpers.go

```diff
@@ -87,3 +102,60 @@ func ParsePageTokenToTime(token string) (time.Time, error) {
 	}
 	return result, err
 }
+
+// GetNextPageTokenFromTime generates a page token. This function should be used when generating a next page token
+// in a List operation on a time.Time field.
+func GetNextPageTokenFromTime(t time.Time) string {
+	if t.IsZero() {
+		return ""
+	}
+	return NewPageToken(t)
+}
+
+// SetCurrentPage generates a set of repository.Option to retrieve results for a specific page.
+func SetCurrentPage(opts []repository.Option, p Pagination) ([]repository.Option, error) {
+	opts = append(opts, firestore.OrderBy(firestore.Ascending("updated_at")))
+	opts, err := setMaxResults(opts, p)
+	if err != nil {
+		return nil, err
+	}
+	if p == nil || len(p.GetPageToken()) == 0 {
+		return opts, nil
+	}
+	updatedAt, err := ParsePageTokenToTime(p.GetPageToken())
+	if err != nil {
+		return nil, err
+	}
+	opts = append(opts, firestore.StartAt(updatedAt))
+	return opts, nil
+}
+
+// setMaxResults establishes the max number of items that should be returned from firestore.
+// This function includes an extra element in the MaxResults option, this last element is used for pagination.
+// The last element should be discarded before the list is returned to the user.
+// See GetListAndCursor for more information.
+func setMaxResults(opts []repository.Option, sg pageSizeGetter) ([]repository.Option, error) {
```

Comment:
nit: this might be a remnant from compilers in other languages being picky about this (looking at you C), but can we always write functions in an order where they are defined before they are used?

In this case: `SetCurrentPage` calls `setMaxResults`, so `setMaxResults` should be defined before `SetCurrentPage`.